app/questionnaire_state/relationship_state_question.py

```diff
@@ -11,13 +12,34 @@ def __init__(self, id, schema_item):
     def build_repeating_state(self, user_input):
         template_answer = self.answers.pop()
         group_instance = template_answer.group_instance
-        household_answers = get_answer_store(current_user).filter(answer_id='household-full-name')
-        sorted(household_answers, key=lambda household_answer: household_answer['answer_instance'])
-        remaining_answers = household_answers[group_instance + 1:] if self.group_instance < len(household_answers) else []
+        first_names = get_answer_store(current_user).filter(answer_id='first-name')
+        last_names = get_answer_store(current_user).filter(answer_id='last-name')
 
-        for index, answer in enumerate(remaining_answers):
+        household_answers = list(zip(first_names, last_names))
+
+        household_members = []
+        for household_answer in household_answers:
+            household_members.append({
+                'first-name': household_answer[0]['value'],
+                'last-name': household_answer[1]['value'],
+            })
+
+        remaining_people = household_members[group_instance + 1:] if self.group_instance < len(household_answers) else []
+
```

Comment:
@ajmaddaford I've made these changes to improve readability. Will defer making the changes to zip longest until our storage mechanism for repeating answers changes.